docs/en/Extensions.md

```diff
@@ -0,0 +1,92 @@
+---
+layout: global
+group: Features
+title: Under Storage Extensions
+priority: 7
+---
+
+* Table of Contents
+{:toc}
+
+Alluxio can be extended with the addition of under storage modules at runtime. Under storage extensions can be built as JARs and included at a specific location to be picked up by core Alluxio without the need to restart. This page describes the internals of how extensions in Alluxio work, and provides instructions for building and installing an extension.
+
+## How it Works
+
+### Service Discovery
+Alluxio servers use Java [ServiceLoader](https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html) to discover implementations of the under storage API. Specifically providers include implementations of the `alluxio.underfs.UnderFileSystemFactory` interface. A common way to advertise an implementation is by including a text file `src/main/resources/META_INF/services/alluxio.underfs.UnderFileSystemFactory` with a single line pointing to the class implementing the said interface.
```

Comment:
- I feel this paragraph is actually implementation rather than "how it works"
I would put it simple here for readers just want to use extension, just saying the extension jars are loaded dynamically at runtime by Alluxio servers, which enables Alluxio to talk to different UFSes